resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/StreamingCompatibilitySuite.scala

```diff
@@ -0,0 +1,220 @@
+        .addToArgs(host, port.toString)
+        .endContainer()
+        .endSpec()
+        .done()
+
+      Eventually.eventually(TIMEOUT, INTERVAL) {
+        assert(getRunLog(driverPodName)
+          .contains("spark-streaming-kube"), "The application did not complete.")
+      }
+    } finally {
+      // Have to delete the service manually since it doesn't have an owner reference
+      kubernetesTestComponents
+        .kubernetesClient
+        .services()
+        .inNamespace(kubernetesTestComponents.namespace)
+        .delete(driverService)
+      serverSocket.close()
+    }
+  }
+
+  test("Run spark streaming in cluster mode.", k8sTestTag) {
+    val (host, port, serverSocket) = startSocketServer()
+    try {
+      runSparkJVMCheckAndVerifyCompletion(
+        mainClass = "org.apache.spark.examples.streaming.NetworkWordCount",
+        appArgs = Array[String](host, port.toString),
+        expectedJVMValue = Seq("spark-streaming-kube"))
+    } finally {
+      serverSocket.close()
+    }
+  }
+
+  test("Run spark structured streaming in cluster mode.", k8sTestTag) {
+    val (host, port, serverSocket) = startSocketServer()
+    try {
+      runSparkJVMCheckAndVerifyCompletion(
+        mainClass = "org.apache.spark.examples.sql.streaming.StructuredNetworkWordCount",
+        appArgs = Array[String](host, port.toString),
+        expectedJVMValue = Seq("spark-streaming-kube"))
+    } finally {
+      serverSocket.close()
+    }
+  }
+
+  test("Run spark structured streaming in client mode.", k8sTestTag) {
+    val (host, port, serverSocket) = startSocketServer()
+    val (driverPort: Int, blockManagerPort: Int, driverService: Service) =
+      driverServiceSetup(driverPodName)
+    try {
+      val driverPod = setupSparkStreamingPod("client", driverPodName)
+        .addToArgs("--conf",
+          s"spark.driver.host=" +
+            s"${driverService.getMetadata.getName}.${kubernetesTestComponents.namespace}.svc")
+        .addToArgs("--conf", s"spark.driver.port=$driverPort")
+        .addToArgs("--conf", s"spark.driver.blockManager.port=$blockManagerPort")
+        .addToArgs("sql.streaming.StructuredNetworkWordCount")
+        .addToArgs(host, port.toString)
+        .endContainer()
+        .endSpec()
+        .done()
+
+      val TIMEOUT = PatienceConfiguration.Timeout(Span(3, Minutes))
+      Eventually.eventually(TIMEOUT, INTERVAL) {
+        assert(getRunLog(driverPodName).contains("spark-streaming-kube"),
+          "The application did not complete.")
+      }
+    }
+    finally {
+      // Have to delete the service manually since it doesn't have an owner reference
+      kubernetesTestComponents
+        .kubernetesClient
+        .services()
+        .inNamespace(kubernetesTestComponents.namespace)
+        .delete(driverService)
+      serverSocket.close()
+    }
+  }
+
+  private def getRunLog(_driverPodName: String): String = kubernetesTestComponents.kubernetesClient
```

Comment:
Why the underscore in `_driverPodName`? If it's a name conflict, I would presume the driver pod name is available elsewhere already and we can just use the local var and not have an argument.